nautobot_golden_config/api/serializers.py

```diff
@@ -68,13 +69,31 @@ class GoldenConfigSettingSerializer(TaggedObjectSerializer, CustomFieldModelSeri
     url = serializers.HyperlinkedIdentityField(
         view_name="plugins-api:nautobot_golden_config-api:goldenconfigsetting-detail"
     )
+    scope = serializers.JSONField(required=False)
+    dynamic_group = NestedDynamicGroupSerializer(required=False)
 
     class Meta:
         """Set Meta Data for GoldenConfigSetting, will serialize all fields."""
 
         model = models.GoldenConfigSetting
         fields = "__all__"
 
+    def validate(self, data):
+        """Validate scope & dynamic_group are not both submitted."""
+        if data.get("scope") and data.get("dynamic_group"):
+            raise serializers.ValidationError("Payload can only contain `scope` or `dynamic_group`, but not both.")
+        return data
+
+    def create(self, validated_data):
+        """Overload to handle ability to post scope instead of dynamic_group."""
+        if not validated_data.get("scope"):
+            return models.GoldenConfigSetting.objects.create(**validated_data)
+        scope = validated_data.pop("scope")
+        setting = models.GoldenConfigSetting(**validated_data)
+        setting.scope = scope
+        setting.save()
```

Comment:
why not `validated_save()`?

Reply:
Data is already validated at that point.